### PR TITLE
refextract: don't raise error when document is xml

### PR DESCRIPTION
**Pull request by MJedr**
ref: inspirehep/inspirehep#2027

### CODE REVIEW

File: inspirehep/modules/workflows/tasks/refextract.py [hunk 1]
```diff
@@ -112,11 +114,16 @@ def extract_journal_info(obj, eng):
...
-        )
+        try:
+            extracted_references = extract_references_from_file(
+                filepath,
+                override_kbs_files=kbs_path,
+                reference_format=u'{title},{volume},{page}',
+            )
+        except UnknownDocumentTypeError as e:
+            if 'xml' in e.message:
+                return
```
Comment (michamos) [inline inspirehep/modules/workflows/tasks/refextract.py:125]: maybe add some `info` logging here?

File: inspirehep/modules/workflows/tasks/refextract.py [hunk 2]
```diff
@@ -112,11 +114,16 @@ def extract_journal_info(obj, eng):
...
-            reference_format=u'{title},{volume},{page}',
-        )
+        try:
+            extracted_references = extract_references_from_file(
+                filepath,
+                override_kbs_files=kbs_path,
+                reference_format=u'{title},{volume},{page}',
+            )
+        except UnknownDocumentTypeError as e:
+            if 'xml' in e.message:
```
[the remark this answers is not in the document]
Comment (MJedr) [reply inspirehep/modules/workflows/tasks/refextract.py:124]: and for 'text/xml' it's ok to raise an exception?

[the remark this answers is not in the document]
Comment (michamos) [reply inspirehep/modules/workflows/tasks/refextract.py:124]: I didn't know we can get both. Then it's ok :+1:

File: inspirehep/modules/workflows/tasks/actions.py [hunk 3]
```diff
@@ -488,7 +488,9 @@ def refextract(obj, eng):
         text_references = dedupe_list(extract_references_from_text(text, source))
         matched_text_references = match_references_based_on_flag(text_references)
 
-    if len(matched_pdf_references) == len(matched_text_references) == 0:
+    if (not matched_pdf_references and not matched_text_references) or (
+        len(matched_pdf_references) == len(matched_text_references) == 0
```
Comment (michamos) [inline inspirehep/modules/workflows/tasks/actions.py:492]: `not [] == True`, so I don't think you need the `len` checks any longer.

**Review by michamos — DISMISSED** [review]
minor comment, looks good otherwise :+1:

**Review by michamos — APPROVED** [review]

**Pull request merged**